src/components/Inputs/input.module.scss

```diff
@@ -1438,6 +1447,129 @@
     &.input-stretch {
         width: 100%;
     }
+
+    &.isExpandable {
+        .expandable-wrapper {
+            position: relative;
+        }
+
+        .input-group {
+            position: absolute;
+            left: 0;
+
+            input {
+                width: 0;
+            }
+        }
+
+        .expandable-thumb {
+            position: absolute;
+            left: -3px;
+        }
+
+        .overlay {
+            position: absolute;
+            left: 0;
+            top: 0;
+            height: 100%;
+            width: 200%;
+            z-index: -1;
+            background: $expandable-header-gradient;
+        }
+
+        &.expand-right {
+            .input-group,
+            .overlay {
+                left: unset;
+                right: 10px;
+            }
+            .overlay {
+                transform: rotate(180deg);
+            }
+        }
+
+        input,
+        .input-group,
+        .overlay {
+            transition: all $motion-duration-extra-fast $motion-easing-easeinout;
+        }
+
+        .input-group:focus-within {
+            @each $size-name, $size-value in $input-size-map {
+                &.input-#{$size-name} {
+                    width: auto;
+                }
+            }
+
+            ~ .action-wrapper {
+                display: block;
+            }
+
+            ~ .expandable-thumb {
+                display: none;
+            }
+        }
+
+        // Set Rules
+        .input-group:hover > input,
+        .input-group:focus-within > input,
+        input:focus,
+        input:not(:placeholder-shown) {
+            @each $size-name, $size-value in $input-size-map {
+                &.input-#{$size-name} {
+                    width: auto;
+                }
+            }
+
+            ~ .action-wrapper {
+                display: block;
+            }
+
+            ~ .expandable-thumb {
+                display: none;
+            }
+        }
+
+        .input-group:hover,
+        .input-group:focus-within {
+            &::after {
+                left: 100%;
+            }
+        }
+
+        // Unset Rules
+        @each $size-name, $size-value in $input-size-map {
+            &.input-#{$size-name} > .expandable-wrapper {
+                min-width: $size-value;
+                min-height: $size-value;
+            }
+
+            .input-group:not(:hover):not(:focus-within)
+                input.input-#{$size-name}:placeholder-shown {
+                min-width: $size-value;
+                padding: 0;
+                overflow: hidden;
+
+                // Internet Explorer 10–11
+                &:-ms-input-placeholder {
```

Comment:
Rather than `font-size`, you might try `opacity` and `width`, then animate them to prevent the overlap when the `input-group `meets the criteria to expand.

Reply:
ahh! I should have done that only. thanks for the trick.

Reply:
`width` may not even be needed as `::placeholder` is a pseudo selector.